extensions/ql-vscode/src/view/data-extensions-editor/MethodRow.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  ExternalApiUsage,
+  ModeledMethod,
+} from "../../data-extensions-editor/interface";
+import {
+  VSCodeDataGridCell,
+  VSCodeDataGridRow,
+  VSCodeDropdown,
+  VSCodeOption,
+  VSCodeTextField,
+} from "@vscode/webview-ui-toolkit/react";
+import * as React from "react";
+import { useCallback, useMemo } from "react";
+import styled from "styled-components";
+
+const Dropdown = styled(VSCodeDropdown)`
+  width: 100%;
+`;
+
+const TextField = styled(VSCodeTextField)`
+  width: 100%;
+`;
+
+type SupportedUnsupportedSpanProps = {
+  supported: boolean;
+};
+
+const SupportedUnsupportedSpan = styled.span<SupportedUnsupportedSpanProps>`
+  color: ${(props) => (props.supported ? "green" : "red")};
+`;
+
+type Props = {
+  method: ExternalApiUsage;
+  model: ModeledMethod | undefined;
+  onChange: (method: ExternalApiUsage, model: ModeledMethod) => void;
+};
+
+export const MethodRow = ({ method, model, onChange }: Props) => {
+  const argumentsList = useMemo(() => {
+    if (method.methodParameters === "()") {
+      return [];
+    }
+    return method.methodParameters
+      .substring(1, method.methodParameters.length - 1)
+      .split(",");
+  }, [method.methodParameters]);
+
+  const handleTypeInput = useCallback(
+    (e: InputEvent) => {
+      const target = e.target as HTMLSelectElement;
+
+      onChange(method, {
+        input: argumentsList.length === 0 ? "Argument[-1]" : "Argument[0]",
+        output: "ReturnType",
+        kind: "value",
+        ...model,
+        type: target.value as ModeledMethod["type"],
```

Comment:
I think we can do this instead

```suggestion
        type: target.value as ModeledMethodType,
```